scripts/monitoring/cron-send-docker-oc-versions.py

```diff
@@ -24,26 +25,60 @@ def parse_args():
     return args
 
 
+def add_specific_rpm_version(package_name, rpm_db_path, keys, mts):
+    '''get rpm package version and add to metric sender and keys dictionary
+    '''
+    try:
+        return_value = subprocess.check_output("rpm --dbpath {rpm_db_path} -q {package}".format(rpm_db_path=rpm_db_path,
+                                                                                                package=package_name),
+                                               stderr=subprocess.STDOUT, shell=True)
+
+        if return_value.startswith(package_name):
+            package_version = return_value[len(package_name)+1:len(return_value)-1]
+            key = "{}.version".format(package_name)
+            mts.add_metric({key: package_version}, key_tags={'miq_metric': 'true'})
+            keys[key] = package_version
+            return True
+
+    except subprocess.CalledProcessError:
+        pass
+
+    return False
+
 def main():
     '''get docker and openshift versions and send to metric sender
     '''
 
     args = parse_args()
     mts = MetricSender(verbose=args.verbose, debug=args.debug)
 
+    # Check if host rpm db is mounted. Otherwise check againts container db
+    rpm_db_path = "/host/var/lib/rpm"
+    if not os.path.exists(rpm_db_path):
+        rpm_db_path = "/var/lib/rpm"
+
+    keys = {}
+
     # Get docker version
-    cli = AutoVersionClient(base_url='unix://var/run/docker.sock', timeout=120)
-    docker_version = cli.version()["Version"]
-    mts.add_metric({"docker.version": docker_version}, key_tags={'miq_metric': 'true'})
+    if not add_specific_rpm_version("docker", rpm_db_path, keys, mts):
+        print "Failed to get docker rpm version."
```

Comment:
This isn't enough of a failure IMO. The script itself needs to exit with a non-zero exit codeo (not necessarily right here, but somewhere in the script).